clients/drcachesim/tools/func_view.cpp

```diff
@@ -61,6 +61,18 @@
 namespace dynamorio {
 namespace drmemtrace {
 
+namespace {
+// Return an indent string for the given indent level.
+std::string
+get_indent_string(int nesting_level)
+{
+    if (nesting_level < 1) {
```

Comment:
But doesn't this indicate some deeper problem, which might surfaces as errors in other parts of the code?  If we don't have time to figure out how it got negative (that would be ideal), wouldn't it be better to either A) make this a fatal error and exit or B) change the level to 0 so nothing else has a problem?  Just changing how it's printed doesn't feel sufficient.  There should at least be a comment explaining whether this is normal or expected.

Reply:
In general I think it should only be a fatal error if state is so hopelessly corrupted that func_view is unable to meaningfully proceed, or if some internal guarantee has been violated.  So when presented with a bad or inconsistent input trace, I think it's best to proceed as normally as possible, though I agree it makes sense to indicate to the user that something seems wrong.  I'll add some checks in the places where nesting_level is decremented and have them warn rather than decrement if it is about to go negative.